src/Core/Util/ConfigurationVariablesLoader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\Dotenv\Dotenv;
+
+class ConfigurationVariablesLoader
+{
+    private $loaded = false;
+
+    private const PARAMETERS_WITH_ENV_VARS = [
+        'database_host' => 'PS_DATABASE_HOST',
+        'database_port' => 'PS_DATABASE_PORT',
+        'database_user' => 'PS_DATABASE_USER',
+        'database_password' => 'PS_DATABASE_PASSWORD',
+        'database_name' => 'PS_DATABASE_NAME',
+        'database_prefix' => 'PS_DATABASE_PREFIX',
+        'database_engine' => 'PS_DATABASE_ENGINE',
+        'cookie_key' => 'PS_COOKIE_KEY',
+        'cookie_iv' => 'PS_COOKIE_IV',
+    ];
+
+    public function loadEnvVariables(array $parameters): array
+    {
+        if (!$this->loaded) {
+            (new Dotenv(true))->loadEnv(_PS_ROOT_DIR_ . '/.env');
+        }
+
+        if (!array_key_exists('parameters', $parameters)) {
+            return $parameters;
+        }
```

Comment:
```suggestion
         if (!isset($parameters['parameters'])) {
            return $parameters;
        }
```

Put the loader in the constructor, with path as parameter.
ALso `$this->loaded` is never used or set, and it's a private property, so if you run this code twice, it generate env variables twice :thinking: